### PR TITLE
[Core] Consistent diagonal setting for constraints in BlockBuilderAndSolver

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

I realize that when setting constraints we always set the maximum value of the diagonal instead of the `mScaleFactor` as in the rest of the code. 

In addition, 2 new access methods are introduced:
- `GetScaleFactor()`: Returns the current value of the scale factor.
- `SetScaleFactor(double ScaleFactor)`: Sets a new value for the scale factor.

**🆕 Changelog**

- [Consistent diagonal setting for constraints](https://github.com/KratosMultiphysics/Kratos/commit/6d7c8c4874fb382126199c93d6e5057c27e0f1ed)
- [Better implementation with avoid unnecessary duplicated computation of the scale factor](https://github.com/KratosMultiphysics/Kratos/pull/10738/commits/cd0c57aac6fafcf2ef785780cedf8f78bead7bfc)
- [Giving up and calculate always](https://github.com/KratosMultiphysics/Kratos/pull/10738/commits/14f50dd02085b90cbe2d086f0455591b52fea9cb)

### CODE REVIEW

**Comment by loumalouomega** [comment]
The builder and solver test fails as expected (I need to update the values), but the eigen tests I don't know why fails, @philbucher do you know these tests?

**Comment by philbucher** [comment]
> The builder and solver test fails as expected (I need to update the values), but the eigen tests I don't know why fails, @philbucher do you know these tests?

no idea
but thats why I am scared of any changes to the B&S such as [this](https://github.com/KratosMultiphysics/Kratos/pull/10706/files#r1096068017)

**Comment by loumalouomega** [comment]
> > The builder and solver test fails as expected (I need to update the values), but the eigen tests I don't know why fails, @philbucher do you know these tests?
> 
> no idea but thats why I am scared of any changes to the B&S such as [this](https://github.com/KratosMultiphysics/Kratos/pull/10706/files#r1096068017)

In that particular case the results obtained are the same as in serial. My question is why the eigen gives a so different value when there are 1s in the diagonal. I will investigate...

**Comment by loumalouomega** [comment]
With the update of the default diagonal values should work now

**Comment by loumalouomega** [comment]
> With the update of the default diagonal values should work now

I notice is not necessarily initialized, so I ensured it is initialized, this way we can avoid recompute the scale factor each time.

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h [hunk 1]
```diff
@@ -1200,6 +1221,7 @@ class ResidualBasedBlockBuilderAndSolver
     std::vector<IndexType> mMasterIds;                /// The equation ids of the master
     std::unordered_set<IndexType> mInactiveSlaveDofs; /// The set containing the inactive slave dofs
     double mScaleFactor = 1.0;                        /// The manually set scale factor
+    bool mScaleFactorInitialized = false;             /// If the value has been initialized
```
Comment (loumalouomega) [inline kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]:
```suggestion

```

**Comment by roigcarlo** [comment]
This also looks good to me. ping rest of the @KratosMultiphysics/technical-committee

**Review by rubenzorrilla — APPROVED** [review]
I agree @roigcarlo. Changes also look correct to me, including the changes in the test values.

**Comment by loumalouomega** [comment]
Now I am going to make consistent the Trilinos one

**Pull request merged**